src/server.tsx

```diff
@@ -376,6 +380,34 @@ app.post('/epic/compare-variant-decision', async (req: express.Request, res: exp
     res.send('thanks');
 });
 
+app.get(
+    '/amp/experiments_data',
+    cors({
+        origin: [
+            'https://amp-theguardian-com.cdn.ampproject.org',
+            'https://amp.theguardian.com',
+            'http://localhost:3030',
+            'https://amp.code.dev-theguardian.com',
+        ],
+        credentials: true,
+        allowedHeaders: ['x-gu-geoip-country-code'],
+    }),
+    async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        try {
+            const response = await getAmpExperimentData();
+
+            // The cache key in fastly is the X-GU-GeoIP-Country-Code header
+            res.setHeader('Surrogate-Control', 'max-age=300');
+            res.setHeader('Cache-Control', 'max-age=60');
```

Comment:
This would be a separate PR, but I'm just thinking that we have to call this quite a lot. I think we could make this more reusable by factoring it out into a middleware i.e.

```js
const cached = (fastlyMaxAge: number, clientMaxAge: number) => (
    req: express.Request,
    res: express.Response,
    next: express.NextFunction,
): void => {
    res.setHeader('Surrogate-Control', `max-age=${fastlyMaxAge}`);
    res.setHeader('Cache-Control', `max-age=${clientMaxAge}`);

    next();
};
```

And then just use it whenever we need it i.e.

```js
app.get(
    '/amp/experiments_data',
    cors({
        origin: [
            'https://amp-theguardian-com.cdn.ampproject.org',
            'https://amp.theguardian.com',
            'http://localhost:3030',
            'https://amp.code.dev-theguardian.com',
        ],
        credentials: true,
        allowedHeaders: ['x-gu-geoip-country-code'],
    }),
    cached(300, 60), // probably want to do something about the magic numbers though...
    async (req: express.Request, res: express.Response, next: express.NextFunction) => {
//...
```

Reply:
what do we reckon @michaelbjacobson , @tomrf1 ?

Reply:
Makes sense to me but, as you say, this is probably best addressed in a separate PR.